Mathlib/Geometry/Manifold/PartitionOfUnity.lean

```diff
@@ -590,3 +621,112 @@ theorem Metric.exists_smooth_forall_closedBall_subset {M} [MetricSpace M] [Chart
   rw [← Metric.emetric_closedBall (hδ0 _).le]
   exact hδ i x hx
 #align metric.exists_smooth_forall_closed_ball_subset Metric.exists_smooth_forall_closedBall_subset
+
+lemma exists_smooth_support_eq_of_model {s : Set H} (hs : IsOpen s) :
+    ∃ f : H → ℝ, f.support = s ∧ Smooth I 𝓘(ℝ) f ∧ Set.range f ⊆ Set.Icc 0 1 := by
+  have h's : IsOpen (I.symm ⁻¹' s) := I.continuous_symm.isOpen_preimage _ hs
+  rcases h's.exists_smooth_support_eq with ⟨f, f_supp, f_diff, f_range⟩
+  refine ⟨f ∘ I, ?_, ?_, ?_⟩
+  · rw [support_comp_eq_preimage, f_supp, ← preimage_comp]
+    simp only [ModelWithCorners.symm_comp_self, preimage_id_eq, id_eq]
+  · exact f_diff.comp_contMDiff contMDiff_model
+  · exact Subset.trans (range_comp_subset_range _ _) f_range
+
+/-- Given an open set in a finite-dimensional real manifold, there exists a nonnegative smooth
+function with support equal to `s`. -/
+theorem exists_smooth_support_eq (hs : IsOpen s) :
```

Comment:
Again I think it would be nice to enable dot notation but I'm less confident of what the name should be. Maybe something like:
```suggestion
theorem IsOpen.exists_mSmooth_support_eq (hs : IsOpen s) :
```